### PR TITLE
enh(LobbySettings): add timezone and relative time

### LINKED ISSUE #12027: Show timezone when setting lobby time
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Create a conversation
2. Enable the lobby
3. Set a start time

### Expected behaviour
It's clear what the timezone the selected time is (even when it's not changable, so don't need a dropdown or something, just say the current timezone that will be used to "math")

### Actual behaviour
grafik
if it's < 24h even visitors have a better experience as they see a relative "in X hours" message.

## Talk app

**Talk app version:** main

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix #12027

### 🖼️ Screenshots / Screencasts

very soon | soon | not in 24 hours |
-- | -- | -- |
image | image | image |

Lobby screen:
image

### 🏁 Checklist

- [ ] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/ConversationSettings/LobbySettings.vue [hunk 1]
```diff
@@ -140,6 +148,33 @@ export default {
...
+			return this.lobbyTimer
+				&& this.lobbyTimer > Date.now()
+				&& (this.lobbyTimer - Date.now()) < ONE_DAY_IN_MS // less than 24 hours
+		},
+
+		getTimeZone() {
+			if (!this.lobbyTimer) {
+				return ''
+			}
+			const date = new Date(this.lobbyTimer)
```
Comment (DorraJaouad) [inline src/components/ConversationSettings/LobbySettings.vue:162]: Aligning with the usage of NcDateTimePicker value as in example
image

File: src/components/ConversationSettings/LobbySettings.vue [hunk 2]
```diff
@@ -140,6 +148,33 @@ export default {
...
+			return t('spreed', `Start time: ${date}`)
+		},
+
+		getRelativeTime() {
+			const diff = this.lobbyTimer - Date.now()
+			const hours = Math.ceil(diff / (60 * 60 * 1000))
+			if (hours > 1) {
+				// TRANSLATORS: hint for the time when the meeting starts
+				return n('spreed', 'in less than %n hour', 'in less than %n hours', hours)
```
Comment (Antreesy) [inline src/components/ConversationSettings/LobbySettings.vue:171]: Maybe use Math.floor for consistency? What's the usual pattern for these cases?

```suggestion
				return n('spreed', 'in %n hour', 'in %n hours', hours)
```

**Review by Antreesy — COMMENTED** [review]
Works good.
But I'd expect to see similar relative numbers in moderator dialog, and lobby waiting page. Maybe we should align the appearance in both places?

image

File: src/components/ConversationSettings/LobbySettings.vue (hunk 2 above)
Comment (DorraJaouad) [reply src/components/ConversationSettings/LobbySettings.vue:171]: I avoided floor because it is misleading IMO: 

left 1h59min -> shows in 1 hour.

Comment (Antreesy) [reply src/components/ConversationSettings/LobbySettings.vue:171]: Same would be in opposite to 1h02min. But anyway, as long as it's consistent in both places (here and at LobbyScreen.vue), I don't mind

Comment (jancborchardt) [reply src/components/ConversationSettings/LobbySettings.vue:171]: We should be exact to minutes here. Can't we say "In 1 hour 24 minutes"?

Comment (jancborchardt) [reply src/components/ConversationSettings/LobbySettings.vue:171]: Short form "In 1h 24m" also works if space is low.

File: src/components/LobbyScreen.vue [hunk 3]
```diff
@@ -94,7 +96,7 @@ export default {
 			if (diff > -45000 && diff < 45000) {
 				return t('spreed', 'The meeting will start soon')
 			}
-			return this.timerInMoment.fromNow()
+			return futureRelativeTime(this.timerInMoment)
```
Comment (Antreesy) [inline src/components/LobbyScreen.vue:99]: You should pass ms, not the Moment object
```suggestion
			return futureRelativeTime(this.timerInMoment.valueOf())
```

Yet it still doesn't show minutes after update (from 6 minutes to 5), can't get why: 
image

**Review by Antreesy — APPROVED** [review]
Works good. Not sure that missing minutes issue is related to changes, as computed value is correct

**Comment by ShGKme** [comment]
> Not sure that missing minutes issue is related to changes, as computed value is correct

It is managed by this server code. If we live-relative-timestampwant to render time manually by Talk, then we need to remove this `live-relative-timestamp` class. Otherwise, I'd say, we should use `nextcloud-vue` component.

https://github.com/nextcloud/server/blob/3ad4bbb09692b180971e024ffeaab1d81fe77e4e/core/src/init.js#L42-L50

**Review by Antreesy — APPROVED** [review]

**Pull request merged**